lcftrans/src/translation.cpp

```diff
@@ -119,8 +118,49 @@ Translation Translation::Merge(const Translation& from) {
 	return stale;
 }
 
-template<typename T, typename U, typename F>
-static void parseEvents(Translation& t, lcf::StringView parent_name, U& root, const F& make_info) {
+using map_event_ctx = const lcf::Context<lcf::rpg::EventCommand,
+		lcf::Context<lcf::rpg::EventPage,
+		lcf::Context<lcf::rpg::Event,
+		lcf::Context<lcf::rpg::Map,
+		lcf::Context<void, void>>>>>;
+
+using common_event_ctx = const lcf::Context<lcf::rpg::EventCommand,
+		lcf::Context<lcf::rpg::CommonEvent,
+		lcf::Context<lcf::rpg::Database,
+		lcf::Context<void, void>>>>;
+
+using battle_event_ctx = lcf::Context<lcf::rpg::EventCommand,
+		lcf::Context<lcf::rpg::TroopPage,
+		lcf::Context<lcf::rpg::Troop,
+		lcf::Context<lcf::rpg::Database,
+		lcf::Context<void, void>>>>>;
```

Comment:
The specialized template func is not invoked when I use a shortened one such as

```
using common_event_ctx = const lcf::Context<lcf::rpg::EventCommand,
		lcf::ContextStructBase<lcf::rpg::CommonEvent>>>;
```

Is this normal?

Reply:
Yes, because that's not the type of the context nor a base class of the context. Putting the base into one of the parent levels produces a different type entirely.

If you wanted to do something like "overload an all EventCommands parented by CommonEvents" then you could do something like:

```c++
// Base case 
template <typename T> 
funcCommonEvent(const T&) {};
// Common Event case
template <typename T> 
funcCommonEvent(Context<EventCommand,Context<CommonEvent<T>>& ctx) {};
```

Templates are essentially pattern matching. Here you're matching the pattern of any Context which has EventCommand and CommonEvent as it's 2 base levels.

Reply:
this works. Thanks. Solved it by adding more :tea:

Reply:
Another way this could work
```c++
template <typename T, typename U>
funcCommonEvent(const T&, const U&) {}

funcCommonEvent(const ContextStructBase<EventCommand>&, const ContextStructBase<CommonEvent>&) {}
```

And call it with
```c++
funcCommonEvent(ctx, *ctx.parent_ctx);
```